commands/core/twig.drush.inc

```diff
@@ -0,0 +1,191 @@
+    // We have this command set to use DRUSH_BOOTSTRAP_DRUPAL SITE. This let's us
+    // run the command and intervene before further bootstrapping into the
+    // _CONFIGURATION and _FULL phases which we do here manually.
+    drush_bootstrap(DRUSH_BOOTSTRAP_DRUPAL_CONFIGURATION);
+
+    // Intervene in settings.php here to skip all database stuff.
+    $settings = Settings::getAll();
+    $settings_object = Settings::getInstance();
+    $ref = new ReflectionProperty($settings_object, 'storage');
+    $ref->setAccessible(TRUE);
+    $settings = drush_twig_custom_settings($settings);
+    $ref->setValue($settings_object, $settings);
+
+    // Add a service provider which will nullify the KeyValue store for us,
+    // because that no longer has a nice convenient $settings override.
+    $GLOBALS['conf']['container_service_providers'][] = new class() implements ServiceProviderInterface, ServiceModifierInterface {
+      public function alter(ContainerBuilder $container) {
+        $container->setParameter('factory.keyvalue', array('default' => 'keyvalue.expirable.null'));
+        $container->setParameter('factory.keyvalue.expirable', array('default' => 'keyvalue.expirable.null'));
+
+        $container->register('config.storage', 'Drupal\Core\Config\NullStorage');
+
+        $container->removeDefinition('database');
+        $container->register('database')->setSynthetic(TRUE);
+      }
+
+      public function register(ContainerBuilder $container) {
+        $container->register('keyvalue.expirable.null', 'Drupal\Core\KeyValueStore\KeyValueNullExpirableFactory');
+      }
+    };
+
+    // Skip the database bootstrapping phase.
+    drush_set_context('DRUSH_BOOTSTRAP_VALIDATION_PHASE', DRUSH_BOOTSTRAP_DRUPAL_DATABASE);
+    drush_set_context('DRUSH_BOOTSTRAP_PHASE', DRUSH_BOOTSTRAP_DRUPAL_DATABASE);
+    // Pretend to have a working site.
+    drush_bootstrap(DRUSH_BOOTSTRAP_DRUPAL_FULL);
+
+    // Now that we're "bootstrapped" we have to fake-up a database connection
+    // so that Twig's service dependency chain can resolve all the way.
+    $container = \Drupal::getContainer();
+    $container->set('database', new class(NULL) extends Connection {
+      public function __construct() {}
+      public function queryRange($query, $from, $count, array $args = array(), array $options = array()) {}
+      public function queryTemporary($query, array $args = array(), array $options = array()) {}
+      public function nextId($existing_id = 0) {}
+      public function mapConditionOperator($operator) {}
+      public function createDatabase($database) {}
+      public function databaseType() {}
+      public function driver() {}
+    });
+  }
+
+  require_once DRUSH_DRUPAL_CORE . "/themes/engines/twig/twig.engine";
+  $twig = \Drupal::service('twig');
+
+  foreach ($searchpaths as $searchpath) {
+    foreach ($file = drush_scan_directory($searchpath, '/\.html.twig/', array('tests')) as $file) {
```

Comment:
Could you do this without `$searchpath`?  Could you just search for all .html.twig classes beneath the Drupal root, for example, without incurring too much of a performance penalty?